### PR TITLE
Feat/laboratoria fe 2/searchbar component

### PR DESCRIPTION
**Pull request by TeresaC21**
## What does this PR do?

Feature about Searchbar component
Implemented: 
  - Consider placeholder text
  - Consider search icon
  - Attach the component to the Navbar

* Investigated and Implemented fn filter input

## Where should the reviewer start?

- Pull branch feat/laboratoria-FE-2/searchbar-component
- Run npm run dev
- Consider on Navbar the component searchbar

:checkered_flag:

## How should this be manually tested?

- Check searchbar when you write on input must show value state React
- Watch list test filter on the main section - Write search and check that filter existent names in the list

## What are the relevant tickets?

 Kanban issue [FE 2] Create Searchbar component
Closes #x

## Questions
 
Please, could you help me to review this PR  @ivanovishado and  @jonathanperazawl?

:question:

### CODE REVIEW

File: components/navbar/navbar.tsx [hunk 1]
```diff
@@ -8,17 +8,22 @@ import LogoBook from '../svg/logoBook';
 
 interface iNavbar {
   color: string;
+  value?: string;
+  setValue?: (e: any) => void;
```
Comment (jonathanperazawl) [inline components/navbar/navbar.tsx]: Avoid using `any`. You can take a look at this article related to react and typescript events
https://fettblog.eu/typescript-react/events/

File: components/searchBar/searchBar.tsx [hunk 2]
```diff
@@ -1,19 +1,49 @@
-import React from 'react';
+import { countReset } from 'console';
```
Comment (jonathanperazawl) [inline components/searchBar/searchBar.tsx]: Are we using this function?

File: components/searchBar/searchBar.tsx [hunk 3]
```diff
@@ -1,19 +1,49 @@
...
+import React, { useState } from 'react';
+
+// Components
+import SearchSVG from '../svg/searchSVG';
 
 interface iSearchbar {
   placeholder: string;
+  value?: string;
+  setValue?: (e: any) => void;
```
Comment (jonathanperazawl) [inline components/searchBar/searchBar.tsx]: Avoid using any for events

File: components/searchBar/searchBar.tsx [hunk 4]
```diff
@@ -1,19 +1,49 @@
...
-const searchBar: React.FC<iSearchbar> = ({ placeholder }) => {
+const searchBar: React.FC<iSearchbar> = ({ placeholder, value, setValue }) => {
+  const handleChange = (e) => {
+    setValue(e.target.value);
+  };
+
+  const handleClick = (e) => {
+    e.preventDefault();
+    setValue(value);
+    setValue('');
```
Comment (jonathanperazawl) [inline components/searchBar/searchBar.tsx:21]: Why are we setting it empty after we replace the value with the same modificator? shouldn't be an effect a better option for this side effect?

**Review by jonathanperazawl — COMMENTED** [review]
Looking good, I left some comments 😄

File: components/searchBar/searchBar.tsx (hunk 3 above)
Comment (TeresaC21) [reply components/searchBar/searchBar.tsx]: Thank you so much for your suggestion, Now I made the changes for interfaces and in the functions passing with the events for react and typescript 

```
interface iSearchbar {
  placeholder: string;
  value: string;
  setValue: (e: string) => void;
}

const searchBar: React.FC<iSearchbar> = ({ placeholder, value, setValue }) => {
  const handleChange = (e: ChangeEvent<HTMLButtonElement>) => {
    setValue(e.target.value);
  };

  const handleClick = (e: FormEvent) => {
    e.preventDefault();
...
...
```

File: components/searchBar/searchBar.tsx (hunk 4 above)
Comment (TeresaC21) [reply components/searchBar/searchBar.tsx:21]: This reason is that after that we write on the input and submit, the input must is clean

**Comment by jonathanperazawl** [comment]
Merged for demo

**Pull request merged**